### PR TITLE
pkg/server: init audit annotations

### PR DESCRIPTION
**Pull request by ibihim**
Signed-off-by: Krzysztof Ostrowski <kostrows@redhat.com>

## Summary

Initializes Audit Annotations with `kaudit.WithAuditAnnotations`.

It is necessary to be done by some early middleware. I verified that with the change we annotate the workspace.

## Related issue(s)
https://github.com/kcp-dev/kcp/issues/862

### CODE REVIEW

**Comment by ibihim** [comment]
```json
{
  "kind": "Event",
  "apiVersion": "audit.k8s.io/v1",
  "level": "Metadata",
  "auditID": "09dc806e-0bd7-49a8-9a56-cf2d3d6603f1",
  "stage": "ResponseComplete",
  "requestURI": "/apis/tenancy.kcp.dev/v1alpha1/clusterworkspaceshards?allowWatchBookmarks=true&resourceVersion=205&timeout=7m7s&timeoutSeconds=427&watch=true",
  "verb": "watch",
  "user": {
    "username": "system:apiserver",
    "uid": "ff888203-3a56-404b-98f6-8bac63fcb546",
    "groups": [
      "system:masters"
    ]
  },
  "sourceIPs": [
    "::1"
  ],
  "userAgent": "Go-http-client/2.0",
  "objectRef": {
    "resource": "clusterworkspaceshards",
    "apiGroup": "tenancy.kcp.dev",
    "apiVersion": "v1alpha1"
  },
  "responseStatus": {
    "metadata": {},
    "code": 200
  },
  "requestReceivedTimestamp": "2022-05-19T18:25:48.489428Z",
  "stageTimestamp": "2022-05-19T18:25:52.725630Z",
  "annotations": {
    "authorization.k8s.io/decision": "allow",
    "authorization.k8s.io/reason": "",
    "tenancy.kcp.dev/workspace": "root"
  }
}
```

File: pkg/server/handler.go [hunk 1]
```diff
@@ -155,11 +155,12 @@ func WithClusterScope(apiHandler http.Handler) http.HandlerFunc {
 func WithClusterAnnotation(handler http.Handler) http.HandlerFunc {
 	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
 		cluster := request.ClusterFrom(req.Context())
+		ctx := kaudit.WithAuditAnnotations(req.Context())
```
Comment (sttts) [inline pkg/server/handler.go:158]: have this here is not good. Please make this explicit in server.go and add a comment why we need it.

Comment (ibihim) [reply pkg/server/handler.go:158]: As a dedicated middleware or as an inline function (to grab hold of the `*request.Request`.

Comment (sttts) [reply pkg/server/handler.go:158]: Somewhere in https://github.com/kcp-dev/kcp/blob/main/pkg/server/server.go#L202, next to your `WithClusterAnnotation`.

**Pull request closed without merging**